bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java

```diff
@@ -288,19 +288,27 @@ protected void processAutomationProvider(Bundle bundle) {
                 if (vendor.getVendorSymbolicName().equals(bundle.getSymbolicName())
                         && !vendor.getVendorVersion().equals(bundle.getVersion().toString())) {
                     List<String> portfolio = providerPortfolio.remove(vendor);
-                    if (portfolio != null && !portfolio.isEmpty())
+                    if (portfolio != null && !portfolio.isEmpty()) {
                         for (String uid : portfolio) {
                             synchronized (providedObjectsHolder) {
                                 providedObjectsHolder.remove(uid);
                             }
                         }
+                    }
                     break;
                 }
             }
         }
-        Enumeration<URL> urlEnum = bundle.findEntries(path, null, false);
-        if (urlEnum == null)
+        Enumeration<URL> urlEnum = null;
+        try {
+            urlEnum = bundle.findEntries(path, null, false);
+        } catch (IllegalStateException e) {
+            logger.error("Can't read from resource of bundle with ID " + bundle.getBundleId(), e);
```

Comment:
Is this really an error situation? I thought that this exception came up on a regular shutdown of the OSGi framework - in such a situation, we should not write error messages to the log.